Userland/Utilities/uniq.cpp

```diff
@@ -5,75 +5,127 @@
  */
 
 #include <AK/RefPtr.h>
+#include <AK/StringView.h>
 #include <LibCore/ArgsParser.h>
+#include <LibCore/Stream.h>
 #include <LibCore/System.h>
-#include <errno.h>
-#include <stdio.h>
-#include <stdlib.h>
-#include <string.h>
 #include <unistd.h>
 
-struct linebuf {
-    char* buf = NULL;
-    size_t len = 0;
-};
+static Optional<String> get_line_content(StringView line, size_t count, bool show_duplicates, bool print_count)
+{
+    if (!show_duplicates || count > 0) {
+        if (print_count)
+            return String::formatted("{} {}\n", count + 1, line);
+        else
+            return String::formatted("{}\n", line);
+    }
+    return {};
+}
 
-static FILE* get_stream(char const* filepath, char const* perms)
+static StringView skip(StringView buf, unsigned int nchars, unsigned int nfields)
 {
-    FILE* ret;
+    StringView tmp(buf);
+    tmp = tmp.trim("\n"sv);
 
-    if (filepath == nullptr) {
-        if (perms[0] == 'r')
-            return stdin;
-        return stdout;
+    if (nfields) {
+        bool in_field = false;
+        auto fields = tmp.find_all([&](char c) {
+            if (c == ' ' || c == '\t' || c == '\v' || c == '\f' || c == '\r')
+                in_field = false;
+            else if (!in_field) {
+                in_field = true;
+                return true;
+            }
+            return false;
+        });
+        if (nfields >= fields.size())
+            nfields = fields.size() - 1;
+        tmp = tmp.substring_view(fields[nfields]);
     }
+    if (tmp.length() < nchars)
+        nchars = tmp.length();
 
-    ret = fopen(filepath, perms);
-    if (ret == nullptr) {
-        perror("fopen");
-        exit(1);
-    }
+    tmp = tmp.substring_view(nchars);
+    return tmp;
+}
 
-    return ret;
+// Borrowed from cmp.cpp, there should probably be a simpler way
```

Comment:
Maybe this means this should be moved into LibCore?

Reply:
Even better, ArgsParser should be able to parse a Core::Stream::File
(e.g.
```cpp
Core::Stream::File in;
args_parser.add_positional_argument(in, "Input file", "input");
```)